src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-multisite-details/rgw-multisite-details.component.ts

```diff
@@ -266,8 +299,58 @@ export class RgwMultisiteDetailsComponent implements OnDestroy, OnInit {
         this.rgwModuleStatus = this.rgwModuleData[0].enabled;
       }
     });
-  }
 
+    this.rgwBucketService
+      .list()
+      .pipe(
+        switchMap((bucketList: any) => {
+          const syncGroupsWithBucketNameObs$ = bucketList.map((bucketName: string) =>
+            this.rgwMultisiteService.getSyncPolicy(bucketName).pipe(
+              map((syncPolicyData: any) => ({
+                bucketName: bucketName,
+                groups: syncPolicyData.groups
+              })),
+              catchError(() => {
+                return of(null);
+              })
+            )
+          );
+          return forkJoin(syncGroupsWithBucketNameObs$);
+        }),
+        catchError(() => {
+          return of([]);
+        })
+      )
+      .subscribe((syncPolicyData: any) => {
+        if (syncPolicyData && syncPolicyData.length > 0) {
+          syncPolicyData.forEach((groupData: any) => {
+            if (groupData.groups.length > 0) {
+              groupData.groups.forEach((group: any) => {
+                this.syncPolicyData.push({
+                  groupName: group.id,
+                  status: group.status,
+                  bucket: groupData.bucketName,
+                  zonegroup: ''
+                });
+              });
+            }
+          });
+        }
+      });
```

Comment:
the only thing that worries me here is the performance of this particular endpoint. The bucket list endpoint already shows stability issues because in a usual customer deployment there are more than 1000s of buckets..

the good thing here is that this doesn't fetch the bucket stats which is usually the less performing thing. But just in case we can try and test with a lot of buckets and see if we see any increase in the request load time.

Reply:
sure, will check by testing more buckets

Reply:
I am thinking to move this logic to backend so that we will just have 1API call to fetch all bucket policies, I wrote some code for it but getting errors

Reply:
moving it to the back-end sounds good to me and that can reduce the load on the browser to some level.. let us know if you are having any issues with it